app/script/conversation/ConversationRepository.coffee

```diff
@@ -1089,9 +1075,23 @@ class z.conversation.ConversationRepository
       .then =>
         if saved_event.type in z.event.EventTypeHandling.STORE
           @_update_message_sent_status conversation_et, saved_event.id
+        @_track_completed_media_action conversation_et, generic_message
 
       return saved_event
 
+  _track_completed_media_action: (conversation_et, generic_message) ->
+    action_type = switch generic_message.content
+      when 'asset' then 'asset' if generic_message.asset.original?
+      when 'knock' then 'ping'
+      when 'text' then 'text' if not generic_message.text.link_preview.length
+
+    return if not action_type
```

Comment:
Use a default `case` in your `switch` construct instead of `return if not action_type`.